[BUG] fixing HMM last read bug

miraep8 opened the pull request:
There was an issue with the way the HMM estimator was setting the last annotated segment in the list.  Thanks to @lmmentel for spotting it/letting me know.  I think this small change should fix it.  Will also add the example provided as a test shortly.

lmmentel commented:
:+1:  might as well fix a typo in this line https://github.com/alan-turing-institute/sktime/blob/3c56f535dc1d060c0053ab6c7f1a1a7199133539/sktime/annotation/hmm.py#L73

miraep8 commented:
Actually just discovered there seems to be another bug, which the test helpfully revealed!  Will fix that and the typo!

miraep8 commented:
In summary - there were actually 2 issues - one was that the last entry was actually being overwritten, and there was also an issue with all the labels being shifted by 1.  I think I have addressed both issues and have also added a sanity check (ie - does HMM work to label a very simple example as expected?)

lmmentel commented on sktime/annotation/hmm.py:
```diff
@@ -374,26 +374,25 @@ def _predict(self, X):
         if self.initial_probs is None:
-            init_probs = 1.0 / (self.num_states) * np.ones(self.num_states)
+            init_probs = 1.0 / self.num_states * np.ones(self.num_states)
         trans_prob, trans_id = HMM._calculate_trans_mats(
```
FYI since you are using `@staticmethod` you can do
```suggestion
        trans_prob, trans_id = self._calculate_trans_mats(
```

miraep8 replied:
Nice :+1 I am not really sure what the difference is between these two things, but seems reasonable to me.  Will look more into it, but in the meantime went ahead and make the change! :)

lmmentel commented:
Nice one @miraep8 :raised_hands: I left a comment in case you want to consider changing `HMM` references to `self` but I'm happy to merge as is.
